aiida/tools/importexport/archive/readers.py

```diff
@@ -0,0 +1,509 @@
+            of the file on which to export.
+        :param sandbox_in_repo: Create the temporary uncompressed folder within the aiida repository
+
+        """
+        super().__init__(filename, **kwargs)
+        self._metadata = None
+        self._data = None
+        # a temporary folder used to extract the file tree
+        self._sandbox: Optional[SandboxFolder] = None
+        self._sandbox_in_repo = sandbox_in_repo
+
+    @property
+    def file_format_verbose(self) -> str:
+        raise NotImplementedError()
+
+    @property
+    def compatible_export_version(self) -> str:
+        return EXPORT_VERSION
+
+    def __enter__(self):
+        super().__enter__()
+        self._sandbox = SandboxFolder(self._sandbox_in_repo)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._sandbox.erase()
+        self._sandbox: Optional[SandboxFolder] = None
+        self._metadata = None
+        self._data = None
+        super().__exit__(exc_type, exc_val, exc_tb)
+
+    def _get_metadata(self):
+        """Retrieve the metadata JSON."""
+        raise NotImplementedError()
+
+    def _get_data(self):
+        """Retrieve the data JSON."""
+        raise NotImplementedError()
+
+    def _extract(self, *, path_prefix: str, progress: bool):
+        """Extract repository data to a temporary folder.
+
+        :param path_prefix: Only extract paths starting with this prefix.
+        :param progress: Whether to report progress of the extraction
+
+        :raises TypeError: if parameter types are not respected
+        """
+        raise NotImplementedError()
+
+    @property
+    def export_version(self) -> str:
+        metadata = self._get_metadata()
+        if 'export_version' not in metadata:
+            raise CorruptArchive('export_version missing from metadata.json')
+        return metadata['export_version']
+
+    @property
+    def metadata(self) -> ArchiveMetadata:
+        metadata = self._get_metadata()
+        export_parameters = metadata.get('export_parameters', {})
+        output = {
+            'export_version': metadata['export_version'],
+            'aiida_version': metadata['aiida_version'],
+            'all_fields_info': metadata['all_fields_info'],
+            'unique_identifiers': metadata['unique_identifiers'],
+            'graph_traversal_rules': export_parameters.get('graph_traversal_rules', None),
+            'entities_starting_set': export_parameters.get('entities_starting_set', None),
+            'include_comments': export_parameters.get('include_comments', None),
+            'include_logs': export_parameters.get('include_logs', None),
+            'conversion_info': metadata.get('conversion_info', [])
+        }
+        try:
+            return ArchiveMetadata(**output)
+        except TypeError as error:
+            raise CorruptArchive(f'Metadata invalid: {error}')
+
+    def entity_count(self, name: str) -> int:
```

Comment:
As mentioned on slack; It’s because they inherit their docstrings from the abstract base class. I would argue that they should not have their own docstrings, since this would essentially be code duplication and would allow the docstring to become out of sync with the base class